src/model_internal.jl

```diff
@@ -0,0 +1,83 @@
+@scoped_enum(
+    RunStatus,
+    NOT_READY = -2,
+    READY = -1,
+    SUCCESSFUL = 0,
+    RUNNING = 1,
+    FAILED = 2,
+)
+
+struct TimeSeriesCacheKey
+    component_uuid::Base.UUID
+    time_series_type::Type{<:TimeSeriesData}
+    name::String
+    multiplier_id::Int
+end
+
+mutable struct SimulationInfo
+    number::Int
+    sequence_uuid::Base.UUID
+end
+
+mutable struct ModelInternal{T <: AbstractOptimizationContainer}
+    container::T
+    ic_model_container::Union{Nothing, T}
```

Comment:
`ic_model_container` and `simulation_info` seem out of place.

Reply:
agree on the simulation info. IC is still needed